### PR TITLE
Align behavior of form- attributes with spec

### PR DESCRIPTION
**Pull request by tuespetre**
### Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### Description

Basically, one of the first tests I wrote for my own project involved reading the `formmethod` attribute on a `<button>`, and I noticed it was just giving me the method on the form, which is... not right. 

This commit makes the `FormAction`, `FormMethod`, etc. properties of `HtmlButtonElement` and `HtmlInputElement` behave in better alignment with the living standard. It also brings some similar tweaks to `HtmlFormElement`.

### CODE REVIEW

**Comment by FlorianRappl** [comment]
Looks good to me, but due to the magnitude of changes more unit tests are necessary for this one.

Please at least provide 2 more unit tests, 1 for valid form action and encoding type and 1 for invalid / fallbacks for these values.

**Comment by tuespetre** [comment]
Word 🗣

I will take a look at breaking down the possible test scenarios for this.

**Comment by FlorianRappl** [comment]
I'll do some of the remaining tests!

**Comment by tuespetre** [comment]
Oh cool... thank you, sorry I am such a scatterbrain.

**Comment by FlorianRappl** [comment]
No problem - all fine! Thanks again for the wonderful PR - much appreciated :beers:
